### ISSUE TITLE
powertop-2.14: compilation fails with 'undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'

### ISSUE THREAD

**Issue by yarda** [fenrus75/powertop#84]
It's using pthread, but it doesn't build/link with -pthread by default. I think it should build out-of-the box without any special user LDFLAGS/CGLAGS. 

Build log snip on Fedora rawhide (f35):
```
libtool: link: g++ -Wall -Wformat -Wshadow -fno-omit-frame-pointer -fstack-protector -I/usr/include/libnl3 -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -Wl,-z -Wl,relro -Wl,--as-needed -Wl,-z -Wl,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -o powertop powertop-devlist.o powertop-display.o powertop-lib.o powertop-main.o calibrate/powertop-calibrate.o cpu/powertop-abstract_cpu.o cpu/powertop-cpu.o cpu/powertop-cpu_core.o cpu/powertop-cpu_linux.o cpu/powertop-cpu_package.o cpu/powertop-cpu_rapl_device.o cpu/powertop-cpudevice.o cpu/powertop-dram_rapl_device.o cpu/powertop-intel_cpus.o cpu/powertop-intel_gpu.o cpu/rapl/powertop-rapl_interface.o devices/powertop-ahci.o devices/powertop-alsa.o devices/powertop-backlight.o devices/powertop-devfreq.o devices/powertop-device.o devices/powertop-gpu_rapl_device.o devices/powertop-i915-gpu.o devices/powertop-network.o devices/powertop-rfkill.o devices/powertop-runtime_pm.o devices/powertop-thinkpad-fan.o devices/powertop-thinkpad-light.o devices/powertop-usb.o measurement/powertop-acpi.o measurement/powertop-extech.o measurement/powertop-measurement.o measurement/powertop-sysfs.o measurement/powertop-opal-sensors.o parameters/powertop-learn.o parameters/powertop-parameters.o parameters/powertop-persistent.o perf/powertop-perf.o perf/powertop-perf_bundle.o process/powertop-do_process.o process/powertop-interrupt.o process/powertop-powerconsumer.o process/powertop-process.o process/powertop-processdevice.o process/powertop-timer.o process/powertop-work.o report/powertop-report-data-html.o report/powertop-report-formatter-base.o report/powertop-report-formatter-csv.o report/powertop-report-formatter-html.o report/powertop-report-maker.o report/powertop-report.o tuning/powertop-bluetooth.o tuning/powertop-ethernet.o tuning/powertop-iw.o tuning/powertop-runtime.o tuning/powertop-tunable.o tuning/powertop-tuning.o tuning/powertop-tuningsysfs.o tuning/powertop-tuningusb.o tuning/powertop-tuningi2c.o tuning/powertop-wifi.o wakeup/powertop-wakeup.o wakeup/powertop-waketab.o wakeup/powertop-wakeup_ethernet.o wakeup/powertop-wakeup_usb.o  -lnl-genl-3 -lnl-3 -lncursesw -lncurses -ltinfo -lpci ../traceevent/.libs/libtraceevnet.a
/usr/bin/ld: /usr/lib/gcc/x86_64-redhat-linux/11/libstdc++.so: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
/usr/bin/ld: /usr/lib64/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```
In the src/Makefile.am it's using PTHREAD_CFLAGS and PTHREAD_LIBS, but it's probably missing corresponding autotools macro call (ax_pthread). I used the following patch in Fedora downstream and can PR if needed:
```
--- a/configure.ac
+++ b/configure.ac
@@ -3,6 +3,7 @@
 
 AC_PREREQ([2.69])
 AC_INIT([powertop], [2.14], [powertop@lists.01.org], [], [https://01.org/powertop])
+AC_CANONICAL_TARGET
 AM_INIT_AUTOMAKE([
 	-Wall
 	1.12.2
@@ -37,6 +38,7 @@ AC_PROG_CC
 AC_PROG_INSTALL
 AM_PROG_CC_C_O
 PKG_PROG_PKG_CONFIG
+AX_PTHREAD
 
 # Checks for libraries.
 
```

**Comment by lionkor** [comment]
I'm getting the same issue on Debian GNU/Linux 10 (buster). 
That patch solved it for me.

**Comment by OS3DrNick** [comment]
I was receiving (almost) the same error on a fresh Ubuntu 20.04 Server,
Making the changes to the configure.ac file made the 'make' command complete successfully.

**Comment by mattst88** [comment]
I think this is fixed by the patches in https://github.com/fenrus75/powertop/pull/82 which are already upstream.

**Comment by yarda** [comment]
> I think this is fixed by the patches in #82 which are already upstream.

Thanks, it builds fine for me, closing.
